hs_access_control/management/commands/community.py

```diff
@@ -0,0 +1,322 @@
+
+    def handle(self, *args, **options):
+
+        if len(options['command']) > 0:
+            cname = options['command'][0]
+        else:
+            cname = None
+
+        if len(options['command']) > 1:
+            command = options['command'][1]
+        else:
+            command = None
+
+        # resolve owner: used in several update commands as grantor
+        if options['owner'] is not None:
+            oname = options['owner']
+        else:
+            oname = 'admin'
+
+        owner = user_from_name(oname)
+        if owner is None:
+            usage()
+            exit(1)
+
+        if options['allow_edit']:  # this is a group privilege
+            privilege = PrivilegeCodes.CHANGE
+        else:
+            privilege = PrivilegeCodes.VIEW
+
+        # not specifing a community lists active communities
+        if cname is None:
+            print("All communities:")
+            for c in Community.objects.all():
+                print("  '{}' (id={})".format(c.name, str(c.id)))
+            exit(0)
+
+        if command is None or command == 'list':
+            community = community_from_name_or_id(cname)
+            if community is None:
+                usage()
+                exit(1)
+
+            print("community '{}' (id={}):".format(community.name, community.id))
+            print("  description: {}".format(community.description))
+            print("  purpose: {}".format(community.purpose))
+            print("  owners:")
+            for ucp in UserCommunityPrivilege.objects.filter(community=community,
+                                                             privilege=PrivilegeCodes.OWNER):
+                print("    {} (grantor {})".format(ucp.user.username, ucp.grantor.username))
+
+            print("  member groups:")
+            for gcp in GroupCommunityPrivilege.objects.filter(community=community):
+                if gcp.privilege == PrivilegeCodes.CHANGE:
+                    others = "can edit community resources"
+                else:
+                    others = "can view community resources"
+                if gcp.allow_view:
+                    myself = 'allows view of group resources'
+                else:
+                    myself = 'prohibits view of group resources'
+                print("     '{}' (id={}) (grantor={}):"
+                      .format(gcp.group.name, gcp.group.id, gcp.grantor.username))
+                print("         {}, {}.".format(myself, others))
+                print("         '{}' (id={}) owners are:".format(gcp.group.name, str(gcp.group.id)))
+                for ugp in UserGroupPrivilege.objects.filter(group=gcp.group,
+                                                             privilege=PrivilegeCodes.OWNER):
+                    print("             {}".format(ugp.user.username))
+            exit(0)
+
+        # These are idempotent actions. Creating a community twice does nothing.
+        if command == 'update' or command == 'create':
```

Comment:
This would be a lot easier to digest if the logic for each command were in it's own method instead of a waterfall of ifs.

Reply:
Do I really have to, mom? Seems that it makes some things harder to understand.